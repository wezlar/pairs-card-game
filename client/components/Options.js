import React, { memo } from 'react'
import PropTypes from 'prop-types'

export function Score (props = {}) {
  const { score } = props

  return (
    <div>
      <div className="score">
        <span className="label">Score: </span>
        <span className="score">{score}</span>
      </div>
      <div className="start-new-game">
        <button>New Game</button>
      </div>
    </div>
  )
}

Score.propTypes = {
  score: PropTypes.number,
}

export default memo(Score)
