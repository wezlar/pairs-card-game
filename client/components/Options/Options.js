import React from 'react';
import PropTypes from 'prop-types';

export function Options (props = {}) {
  const { score, isGameComplete, startNewGame } = props;

  return (
    <div className="options">
      { isGameComplete && <div className="game-complete">
          Yay! All cards matched!
        </div>
      }
      <div className="score">
        <span className="label">Score: </span>
        <span className="score">{score}</span>
      </div>
      <div className="start-new-game">
        <button className="start-new-game__button" onClick={startNewGame}>New Game</button>
      </div>
    </div>
  );
}

Options.propTypes = {
  score: PropTypes.number,
  isGameComplete: PropTypes.bool,
  startNewGame: PropTypes.func,
};

export default Options;
