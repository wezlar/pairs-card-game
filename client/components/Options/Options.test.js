import React from 'react'
import { shallow } from 'enzyme'
import { Options as Component } from './Options';

describe(`<Card />`, () => {
  test(`It doesn't blow up`, () => {
    const tree = shallow(<Component />);
    expect(tree.length).toEqual(1);
  });

  const mockOnClick = jest.fn();
  const props = {
    isGameComplete: false,
    score: 10,
    startNewGame: mockOnClick,
  };

  test(`Renders correctly`, () => {
    const options = shallow(<Component { ...props } />);
    expect(options.debug()).toMatchSnapshot();
  });

  test(`Click is triggered`, () => {
    const options = shallow(<Component { ...props } />);
    expect(mockOnClick).not.toHaveBeenCalled();
    options.find('.start-new-game__button').simulate('click');
    expect(mockOnClick).toHaveBeenCalled();
    options.find('.start-new-game__button').simulate('click');
    expect(mockOnClick).toHaveBeenCalledTimes(2);
  })
})