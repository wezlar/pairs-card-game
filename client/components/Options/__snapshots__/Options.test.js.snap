// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Renders correctly 1`] = `
"<div className=\\"options\\">
  <div className=\\"score\\">
    <span className=\\"label\\">
      Score: 
    </span>
    <span className=\\"score\\">
      10
    </span>
  </div>
  <div className=\\"start-new-game\\">
    <button className=\\"start-new-game__button\\" onClick={[Function: mockConstructor]}>
      New Game
    </button>
  </div>
</div>"
`;
